### DISCUSSION TITLE
Branches naming conventions

### DISCUSSION THREAD

**Discussion by ivavay** [mdwiltfong/secret-santa#24] [Q&A] [answered]
I was wondering what are the naming conventions for branches? And do I need to make a new branch every time I make commits (even if they're tiny changes) to my repo (fork)?

**Comment by mdwiltfong** [comment, marked as answer]
That's a good question. I'm going to answer this in parts. 

- Typically I reference the issue number in the branch name plus my gh username. That way, when people are looking through the branches of the repo, they see what this PR is for, and who made it.
- Making a new branch for every commit is way too much. When you make a branch, it's to create a pull request to resolve an issue. You can think of it as telling a story. What story does this branch/pr tell? If you your branch/pr is too convoluted then it will be harder for people to review it. For many reasons. One, they'll probably have too many questions that may involve context switching, and if your PR touches a lot of pieces of code, then suddenly the repo may require large updates to accommodate the PR (new e2e tests, new gh action workflows, etc). In your case, you are doing a lot of frontend work in this repo. The PR we reviewed recently is probably getting too bloated at this point. Mainly because it already offers a lot to the repo. It offers a structure to how the frontend will work. 

Hopefully that makes sense. There's no sure answer to it. I would say try to isolate your PRs to certain issues. If there isn't an issue that exists for the PR you're making, then you should probably make one and go from there.

**Reply by ivavay** [reply]
Okay I think that makes sense! Thanks for the detailed explanation. Then in my case, maybe I should've created smaller issues from the overall frontend/mockups issue to work off of 😅
